healthcheck.sh

```diff
@@ -0,0 +1,238 @@
+#!/bin/bash
+#
+# Healthcheck script for MariaDB
+#
+# If all of the requested health checks as arguments succeed, then its healthy
+# otherwise its not.
+#
+# Arguments are processed in strict order. Set replication_* options before
+# The --replication check. This allows a different set of replication checks
+# on different connections.
+#
+# --su{=|mysql} is option to run the healthcheck as a different unix user.
```

Comment:
Do we really need this? IMHO we should always `gosu mysql "${BASH_SOURCE[0]}" "$@"` when run by `root` and fail otherwise. I can't think of a reason to use another user. You got a use case in mind? :thinking: 

For the tests that require authentication we should simply require #409, as adding `--user`, `--password`, etc. isn't feasible IMHO.

Reply:
> `MARIADB_ALLOW_EMPTY_ROOT_PASSWORD=1` for testing, user/password in a `$ROOT/.my.cnf` or under `[client]` in the usual configuration mount point, and adding `--defaults-*` per below.

Ah, I see, yes, indeed. Forcing `gosu mysql` for root and failing for others might indeed be a bad idea then.

As an alternative we could drop `gosu` altogether, i.e. requiring users to use `--health-cmd="gosu my_user healthcheck.sh …"` instead. However, I wouldn't say that this is the best way to go, I just want to put everything on the table :thinking: IMO both solutions are valid and the decision is up to you.

> Other options about providing authentication would be about populating configuration files with user/password which in some circumstances might be ok. This means I should accept the `--defaults-{file,extra-file,group-suffix}` for this script as well and then people can keep their heathcheck credentials separate.

:+1: